### PR TITLE
Phase7 minify

### PR DESCRIPTION
**Pull request by cxoltero**
Branch is working as expected.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -21,5 +21,26 @@
...
+      "dependencies": {
+        "jquery": "2.1.4"
+      }
+    },
+    "bootstrap": {
+      "main": [
+        "dist/js/bootstrap.js",
+        "dist/css/bootstrap.css"
+      ]
+    }
```
Comment (paul-barry-kenzan) [inline bower.json:44]: Nice job here!

File: gulpfile.js [hunk 2]
```diff
@@ -2,17 +2,82 @@
...
 gulp.task('css:lint', function() {
   return gulp
-      .src('./src/css/**/*.js')
-      .pipe(validateCssPipeline.validateCSS());
+    .src('./src/css/**/*.css')
+    .pipe(validateCssPipeline.validateCSS());
+});
+
+//concatenate files for venders and app
+gulp.task('js:concatBower', function () {
```
Comment (paul-barry-kenzan) [inline gulpfile.js]: Task names are typically lowercase. So in this case, how do you feel about renaming to something like `js:concat:bower`?

File: gulpfile.js [hunk 3]
```diff
@@ -2,17 +2,82 @@
...
+});
+gulp.task('js:concatApp', function() {
+  return gulp
+    .src(['!./src/js/**/*.test.js', './src/js/**/*.js'])
+    .pipe(concat('app.min.js'))
+    .pipe(gulp.dest('./dest/js/'));
+});
+
+//collect required assets
+gulp.task('getAssets', function () {
```
Comment (paul-barry-kenzan) [inline gulpfile.js]: If this task simply copies files from one folder to another, let's use build-friendly terms. Could we rename this to something like: `copy:assets`?

File: package.json [hunk 4]
```diff
@@ -20,6 +20,12 @@
   "devDependencies": {
     "chai": "2.3.0",
     "dirty-chai": "1.2.2",
+    "gulp-concat": "^2.6.0",
+    "gulp-concat-css": "^2.2.0",
+    "gulp-cssnano": "^2.1.0",
+    "gulp-html-replace": "^1.5.5",
+    "gulp-minify": "0.0.5",
+    "gulp-uglify": "^1.5.1",
```
Comment (paul-barry-kenzan) [inline package.json]: The version numbers should be locked down.

File: pipelines/build.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+var gulp = require('gulp');
+
+gulp.task("")
```
Comment (paul-barry-kenzan) [inline pipelines/build.js]: No need to check in this broken out files yet. Please remove from the PR.

File: gulpfile.js (hunk 3 above)
Comment (cxoltero) [inline gulpfile.js]: it copies the images from one folder, but also replaces the script tags and link tags in the html file. as well as rewrite the html file that we have into a newone with the new tags.

**Comment by paul-barry-kenzan** [comment]
The only suggestion would be to delete the `/dest` directory before copying everything over. This ensures that everything inside `dest/` is fresh and not leftover.

**Pull request merged**